### ISSUE TITLE
Overrides don't get applied to var-definitions during analysis

### ISSUE THREAD

**Issue by sheluchin** [clj-kondo/clj-kondo#1758]
Version: v2022.06.22 

The [overrides](https://github.com/clj-kondo/clj-kondo/blob/master/src/clj_kondo/impl/overrides.clj#L22-L25) are being applied to var usages but not var definitions during analysis. My understanding is that it works for usages because the call `(l/lint-var-usage ctx (override idacs))` ends up calling `analysis/reg-usage!` with the overrides applied, whereas `analysis/reg-var!` only receives the `analysis` map.

https://github.com/clj-kondo/clj-kondo/blob/master/src/clj_kondo/core.clj#L199-L208

You can see the inconsistency by comparing the definition and usage of `defn-`, which is defined in `overrides.clj` with a `:varargs-min-arity 2`:

```clojure
;; definition
{:ns clojure.core
 :name defn-
 :doc "same as defn, yielding non-public def"
 :varargs-min-arity 1
 :arglist-strs ["[name & decls]"]}

;; usage
{:name defn-,
 :filename "/tmp/foo.clj",
 :from foo,
 :macro true,
 :col 2,
 :arity 3,
 :varargs-min-arity 2,
 :row 8,
 :to clojure.core}
```

You can also reproduce this in your own namespace as long as you add the entry to `overrides.clj`:

```clojure
(ns foo)

(defn- bar [x & y])

(bar :_ :_)
```

```clojure
;; overrides.clj
(assoc-in '[:clj :defs foo bar] '{:ns foo
                                  :name bar
                                  :varargs-min-arity 2})
```

```clojure
{:var-definitions
 [{:end-row 3,
   :name-end-col 11,
   :name-end-row 3,
   :private true,
   :name-row 3,
   :ns foo,
   :name bar,
   :defined-by clojure.core/defn-,
   :col 1,
   :name-col 8,
   :end-col 20,
   :arglist-strs ["[x & y]"],
   :varargs-min-arity 1,
   :row 3}],
 :var-usages
 [{:end-row 3,
   :name-end-col 7,
   :name-end-row 3,
   :name-row 3,
   :name defn-,
   :from foo,
   :macro true,
   :col 1,
   :name-col 2,
   :end-col 20,
   :arity 2,
   :varargs-min-arity 2,
   :row 3,
   :to clojure.core}
  {:end-row 5,
   :name-end-col 5,
   :name-end-row 5,
   :name-row 5,
   :name bar,
   :from foo,
   :col 1,
   :name-col 2,
   :end-col 12,
   :arity 2,
   :varargs-min-arity 2,
   :row 5,
   :to foo}]}
```

My expectation is that overrides get applied to both definitions in a consistent manner, such that the records can be joined without conflict.

I took a little while to try to fix it, but I'm afraid I'm not familiar enough with the clj-kondo code to make a change of this size. There are some comments in place already about making improvements around the handling of overrides and about changing the structure to have the linter called from analysis rather than the other way around as it is now. I hope that by filing this issue the task of fixing it is made easier for someone more familiar with kondo than I am.

**Comment by borkdude** [comment]
Thanks

**Comment by borkdude** [comment]
@sheluchin I made issue #1764 which I think is better to solve first as issues like this would then be automatically solved.

**Comment by borkdude** [comment]
I fixed issue #1764 but it doesn't fix the above issue since `reg-var!` does not read anything from the cache.

I'm open to a PR which shares the data between the overrides and reg-var so the data becomes more consistent.

**Comment by sheluchin** [comment]
Thank you, @borkdude. It's a step in the right direction.

I'm a little busy right now but I will try to put together a PR when I get a little more time.
